src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -994,4 +1133,18 @@ grpc_error* ConnectedSubchannel::CreateCall(const CallArgs& args,
   return GRPC_ERROR_NONE;
 }
 
+size_t ConnectedSubchannel::GetInitialCallSizeEstimate(
+    size_t parent_data_size) const {
+  size_t allocation_size =
+      GPR_ROUND_UP_TO_ALIGNMENT_SIZE(sizeof(grpc_subchannel_call));
+  if (parent_data_size > 0) {
+    allocation_size +=
+        GPR_ROUND_UP_TO_ALIGNMENT_SIZE(channel_stack_->call_stack_size) +
+        parent_data_size;
+  } else {
+    allocation_size += channel_stack_->call_stack_size;
```

Comment:
I understand the code used to be like this but why it is rounded up in line 1142 but not here?

Reply:
The alignment padding is only needed when allocating space for two objects at the same time, where the address of the second object needs to start at the alignment boundary.  In the case where `parent_data_size` is zero, there is no second object, so no padding is needed.

Reply:
I see. Thanks!